### PR TITLE
Fix windows Wix creation / json file + signing/cab file paths + configurable NativeMessageHost filename

### PR DESCRIPTION
**Pull request by AndyTather**
Just a few little tweaks to get a new windows sample up and compiling.
Make NativeMessageHost name configurable

### CODE REVIEW

**Comment by taxilian** [comment]
Hmm; some of this has been superceded by other things I've done lately. Let me get things stabilized and I'll look at it again.  Sorry :-/

**Comment by AndyTather** [comment]
Added a commented out template for extension installing from chrome webstore

File: src/ActiveXCore/ActiveXBrowserHost.cpp [hunk 1]
```diff
@@ -111,6 +111,39 @@ int FB::ActiveX::ActiveXBrowserHost::delayedInvoke(const int delayms, const FB::
...
+		IDispatchAPIPtr win = std::dynamic_pointer_cast<IDispatchAPI>(window->getJSObject());
+		helper = std::dynamic_pointer_cast<IDispatchAPI>(win->GetPropertySync("FireBreathPromise").cast<JSObjectPtr>());
+	}
+	return helper;
+}
+
+IDispatchAPIPtr ActiveXBrowserHost::getPromiseProperty(IDispatchAPIPtr obj, std::string name)
+{
+	// ISSUE: not finding returning item?
+	boolean flag = obj->HasProperty(name);
```
Comment (AndyTather) [inline src/ActiveXCore/ActiveXBrowserHost.cpp]: Next call isn't returning promise property

File: src/ActiveXCore/ActiveXBrowserHost.cpp [hunk 2]
```diff
@@ -111,6 +111,39 @@ int FB::ActiveX::ActiveXBrowserHost::delayedInvoke(const int delayms, const FB::
...
+	IDispatchAPIPtr helper = std::dynamic_pointer_cast<IDispatchAPI>(obj->GetPropertySync(name).cast<JSObjectPtr>());
+
```
Comment (AndyTather) [inline src/ActiveXCore/ActiveXBrowserHost.cpp]: Above call isn't returning promise property

File: src/ActiveXCore/JSAPI_IDispatchEx.h [hunk 3]
```diff
@@ -404,9 +411,10 @@ namespace FB { namespace ActiveX {
                     return E_INVALIDARG;
 
                 // TODO: Create a promise from the DOM, return it tied to the Deferred returned here
-                //FB::variant rVal = api->GetProperty(wsName);
+                FB::variantPromise rVal = api->GetProperty(wsName);
+//                host->getComVariant(pvarRes, rVal);
 
-                //host->getComVariant(pvarRes, rVal);
+				setPromise(rVal, pvarRes);
```
Comment (AndyTather) [inline src/ActiveXCore/JSAPI_IDispatchEx.h]: Trying to reuse setPromise pattern/call as in npapi

File: fbgen/src/PluginConfig.cmake [hunk 4]
```diff
@@ -18,6 +18,8 @@ set(PLUGIN_CRX_KEY "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDOqMBmB8ThMqFT3/9s/bfS
 set(PLUGIN_CRX_ID "dlfdfkalmjbghacnadegnkmbcnpmnmfn")
 set(PLUGIN_CRX_WYRMHOLE_NAME ${COMPANY_NAME})
 set(PLUGIN_CRX_NATIVEHOST_NAME "${COMPANY_REVDOMAIN}.firewyrmhost")
+set(PLUGIN_CRX_NATIVEHOST_DESCRIPTION "${COMPANY_NAME} FireWyrm Native Messaging Wyrmhole")
```
Comment (AndyTather) [inline fbgen/src/PluginConfig.cmake]: native host json manifest is rejected by chrome for not having a description.

File: fbgen/src/PluginConfig.cmake [hunk 5]
```diff
@@ -18,6 +18,8 @@ set(PLUGIN_CRX_KEY "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDOqMBmB8ThMqFT3/9s/bfS
...
+set(FBSTRING_HostFileName "FireWyrmNativeMessageHost")
```
Comment (AndyTather) [inline fbgen/src/PluginConfig.cmake]: Wix installer needs a custom name set as FBSTRING_HostFileName but is only set in the nativemessagehost make script, if not here.

**Comment by AndyTather** [comment]
outdated

**Pull request closed without merging**